inferno-core/src/Inferno/Infer.hs

```diff
@@ -379,7 +388,7 @@ inferTypeReps allTypeClasses (ForallTC tvs tyCls (ImplType _impl ty)) inputTys o
   let cs =
         [Right (dummyPos, c) | c@(TypeClass nm _) <- Set.toList tyCls, nm /= "rep"]
           ++ mkConstraints ty inputTys
-   in case runSolve allTypeClasses cs of
+   in case runSolve 0 allTypeClasses cs of -- TODO what if 't0 is in the type passed as argument to this function?
```

Comment:
@goodlyrottenapple would I need to find the largest type var in the constraints `cs` and set the initial `varCount` to that plus one?

Reply:
Yes, i think so